samcli/lib/build/bundler.py

```diff
@@ -0,0 +1,143 @@
+"""
+Handles bundler properties as needed to modify the build process
+"""
+import logging
+from copy import deepcopy
+from typing import Dict, Optional
+
+from samcli.lib.providers.provider import Stack
+from samcli.lib.providers.sam_function_provider import SamFunctionProvider
+
+LOG = logging.getLogger(__name__)
+
+ESBUILD_PROPERTY = "esbuild"
+
+
+class EsbuildBundlerManager:
+    def __init__(self, stack: Stack, template: Optional[Dict] = None):
+        self._stack = stack
+        if template is None:
+            template = dict()
+        self._previous_template = template
```

Comment:
nit/question: this vs `self._previous_template = template or dict()`?

Reply:
Updated.